dist-build/emscripten.sh

```diff
@@ -62,9 +63,32 @@ emmake make clean
 [ $? = 0 ] || exit 1
 
 if [ "$DIST" = yes ]; then
-  emmake make $MAKE_FLAGS install && \
-  emcc "$CFLAGS" --llvm-lto 1 --memory-init-file 0 $CPPFLAGS $LDFLAGS $JS_EXPORTS_FLAGS \
-    "${PREFIX}/lib/libsodium.a" -o "${PREFIX}/lib/libsodium.js" || exit 1
+  emccLibsodium () {
+    outFile="${1}"
+    shift
+    emcc "$CFLAGS" --llvm-lto 1 $CPPFLAGS $LDFLAGS $JS_EXPORTS_FLAGS ${@} \
+      "${PREFIX}/lib/libsodium.a" -o "${outFile}" || exit 1
+  }
+  emmake make $MAKE_FLAGS install || exit 1
+  emccLibsodium "${PREFIX}/lib/libsodium.asm.tmp.js" -Oz -s RUNNING_JS_OPTS=1 -s NO_EXIT_RUNTIME=1
+  emccLibsodium "${PREFIX}/lib/libsodium.wasm.tmp.js" -O3 -s WASM=1
+
+  cat > "${PREFIX}/lib/libsodium.js" <<- EOM
+    if (typeof Module === 'undefined') {
+      var Module = {};
+    }
+    Module.ready = new Promise(function (resolve, reject) {
+      Module.onAbort = reject;
+      Module.onRuntimeInitialized = function () { resolve(); };
```

Comment:
Oops, this should just be `Module.onRuntimeInitialized = resolve;`. Originally it was running `resolve(Module)` before I changed a few things, but now it's slightly unnecessarily verbose. Won't submit a new PR just for that now that it's been merged, but just pointing it out for your reference @jedisct1.